WordPress/gutenberg
# Add reusable BlockPopover and BlockPopoverInbetween components, remove default block appender button for empty paragraphs

Extracted from #40376

## What?

This is a refactor PR. It extracts two generic popover components that allows us to add UI to the block editor that is contextual to given blocks.

## Why?

Right now these components are used to display the block toolbar and the in-between inserter. They'll be used for more in #40376 

We can also potentially leverage them for more (thinking about the BoxVisualizer for instance here)

**Notes**

- After discussing with @jasmussen this PR also removes the inserter that was being shown at the right of selected empty paragraphs.

## Testing Instructions

There should be no behavior/visual change aside the removal of the side inserter in paragraph blocks. So check block toolbars, block inserters, in between inserters.
